app/code/Magento/Customer/Test/Unit/Ui/Component/Listing/Column/GroupActionsTest.php

```diff
@@ -0,0 +1,253 @@
+ */
+class GroupActionsTest extends TestCase
+{
+    /**
+     * @var GroupActions
+     */
+    private $component;
+
+    /**
+     * @var ContextInterface|MockObject
+     */
+    private $contextMock;
+
+    /**
+     * @var UiComponentFactory|MockObject
+     */
+    private $uiComponentFactoryMock;
+
+    /**
+     * @var UrlInterface|MockObject
+     */
+    private $urlBuilderMock;
+
+    /**
+     * @var Escaper|MockObject
+     */
+    private $escaperMock;
+
+    /**
+     * @var GroupManagementInterface|MockObject
+     */
+    private $groupManagementMock;
+
+    /**
+     * Set Up
+     */
+    public function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+
+        $this->contextMock = $this->getMockBuilder(ContextInterface::class)->getMockForAbstractClass();
+        $this->uiComponentFactoryMock = $this->createMock(UiComponentFactory::class);
+        $this->escaperMock = $this->createMock(Escaper::class);
+        $this->groupManagementMock = $this->createMock(GroupManagementInterface::class);
+        $this->urlBuilderMock = $this->getMockForAbstractClass(
+            UrlInterface::class,
+            [],
+            '',
+            false
+        );
+
+        $this->component = $objectManager->getObject(
+            GroupActions::class,
+            [
+                'context' => $this->contextMock,
+                'uiComponentFactory' => $this->uiComponentFactoryMock,
+                'urlBuilder' => $this->urlBuilderMock,
+                'escaper' => $this->escaperMock,
+                'components' => [],
+                'data' => [
+                    'name' => 'name'
+                ],
+                'groupManagement' => $this->groupManagementMock
+            ]
+        );
+    }
+
+    /**
+     * Test data source with a non default customer group
+     *
+     * @dataProvider customerGroupsDataProvider
+     *
+     * @param array $items
+     * @param bool $isDefaultGroup
+     * @param array $expected
+     */
+    public function testPrepareDataSourceWithNonDefaultGroup(array $items, bool $isDefaultGroup, array $expected)
+    {
+        $customerGroup = 'General';
+        $dataSource = [
+            'data' => [
+                'items' => $items
+            ]
+        ];
+        $expectedDataSource = [
+            'data' => [
+                'items' => $expected
+            ]
+        ];
+
+        $this->groupManagementMock->expects($this->any())
+            ->method('isReadonly')
+            ->with(1)
+            ->willReturn($isDefaultGroup);
+        $this->escaperMock->expects($this->any())
+            ->method('escapeHtml')
+            ->with($customerGroup)
+            ->willReturn($customerGroup);
+        $this->urlBuilderMock->expects($this->any())
+            ->method('getUrl')
+            ->willReturnMap(
+                [
+                    ['customer/group/edit', ['id' => 1], 'http://magento.com/customer/group/edit'],
```

Comment:
Replace magic numbers with `const` named for example - `STUB_LOGGEDIN_CUSTOMER_GROUP_ID`
Instead of literal with sample domain - try to also replace it with `const` - eg. `STUB_GROUP_EDIT_URL`